### DISCUSSION TITLE
Make a web assembly

### DISCUSSION THREAD

**Discussion by seyeonHello** [wegylexy/rnnoise_wasm#13] [Q&A] [answered]
Hello! your source code is amazing!
But I have some questions.
I downloaded your source code and executed it myself. (I compiled the source code for "xiph/rnnoise" and downloaded emscripten too) 
And **I ran your "build.sh" file below as it is.**

`emcc \
    -s ENVIRONMENT=worker \
    -s TOTAL_STACK=49152 -s TOTAL_MEMORY=327680 \
    -g0 -O3 --no-entry -Wno-null-dereference \
    -o dist/rnnoise-processor.wasm \
    -Irnnoise/include \
    rnnoise/src/celt_lpc.c \
    rnnoise/src/denoise.c \
    rnnoise/src/kiss_fft.c \
    rnnoise/src/pitch.c \
    rnnoise/src/rnn.c \
    rnnoise/src/rnn_data.c \
    src/worklet.c
`

A wasm file was created, but the "vadprob" value is strange. It is performed differently from the wasm you uploaded. Your wasm works well with vadprob value. But mine is not good. (I can hear fine noises.)
**Do I have to modify the "build.sh" file or "worklet.c" file?**
I wait for a reply.

**Comment by wegylexy** [comment, marked as answer]
Short answer, no.

If you `git clone`d with `--recurse-submodules`, you would get the exact version of `xiph/rnnoise` we used. If you download them separately without `git`, I have no idea how different your copies are from ours.

I guess they have changed `celt_lpc.c` recently, if that matters.

**Reply by wegylexy** [reply]
If you read `worklet.c`, it is just a simple wrapper... https://github.com/wegylexy/rnnoise_wasm/blob/776ca4945d4c0f51212c56d0652c9c3195a48661/src/worklet.c#L57

and `build.sh` simply compiles and links.

**Reply by seyeonHello** [reply]
Thank you for your answer:) I will attempt your guide. For now I lowered the emsdk version and solved it.

**Reply by wegylexy** [reply]
Do you know the docker tag of that version?

**Reply by seyeonHello** [reply]
Sorry I thought the comment section was closed. 
I didn't use docker. I used emscripten version 2.0.3
